sstables/sstables_manager.hh

```diff
@@ -74,6 +75,13 @@ public:
     future<> stop();
 };
 
+struct sstables_manager_subscriber {
+    virtual ~sstables_manager_subscriber() {}
+
+    virtual future<> on_add(generation_type) { return make_ready_future(); };
+    virtual future<> on_unlink(generation_type) { return make_ready_future(); };
```

Comment:
Maybe better to make the notifications non-yielding and let the subscriber handle the complexity.

Reply:
okay

Reply:
kept callbacks as returning future<> in v2 ([06da3f3](https://github.com/scylladb/scylladb/commit/06da3f30f0c2da5e684fbff7b517e97d70d70559))
that is not too complicated with boost::signals2